### ISSUE TITLE
copy items are not saved, log: "Clipboard unchanged, selection unchanged. Test again in 0ms."

### ISSUE THREAD

**Issue by arbach** [hluk/CopyQ#1505]
**the bug**
copyq starts and works properly on ubuntu, randomly after a day or two without restarting, maybe after suspend, copyq stops saving the clipboard entries. 
The issue is only resolved after restart. Stopping and starting copyq doesn't resloved it. even if the configuration folder is completely removed

**To Reproduce**

1. start copyq on ubuntu
2. wait few days without restarting
3. Copy anything to clipboard, using any tools, mouse, keyboard or xclip
4. The item is not saved in copyq

Everything in the log seems normal, except nothing happen when copying to clipboard. 

**Version, OS and Environment**
Ubuntu 20.04 LTS

**Additional context**
Tried disable and enabled "Clipbaoard storing" did not help 

When watching the logs, I got the following when I click "ctl+c"

: Clipboard unchanged, selection unchanged. Test again in 0ms.

Debug logo

`DEBUG: Geometry: Window "MainWindow": Restore geometry "Options/MainWindow_geometry": 645x421,673,343 -> 645x421,267,301
DEBUG: Focus window is " _hh_ "
DEBUG: Loading configuration
DEBUG: Configuration loaded
DEBUG: Starting monitor
DEBUG: Executing: copyq --clipboard-access monitorClipboard
DEBUG: Starting callback: onStart
DEBUG: Executing: copyq onStart
DEBUG: Got save state request from session manager.
DEBUG: Focus window is " _hh_ "
DEBUG: Updating tray menu
DEBUG: Tab "&clipboard": Loading items
DEBUG: Tab "&clipboard": 2338 items loaded
DEBUG: Executing: copyq eval -- runMenuCommandFilters()
DEBUG: Updating tray menu
DEBUG: Focus window is "copyq not saving clipboard - Google Search - Brave"
DEBUG: Geometry: Window "ConfigurationManager": Restore geometry "Options/ConfigurationManager_geometry": 679x621,656,243 -> 1049x621,327,211
DEBUG: Focus window is "Configuration — CopyQ"
DEBUG: Focus window is "copyq not saving clipboard - Google Search - Brave"
DEBUG: Terminating monitor
DEBUG: Executing: copyq eval -- setTitle(); showDataNotification()
DEBUG: Clipboard monitoring disabled.
DEBUG: Monitor finished
DEBUG: Raising window "copyq not saving clipboard - Google Search - Brave"
DEBUG: Focus window is "copyq not saving clipboard - Google Search - Brave"
DEBUG: Focus window is "copyq not saving clipboard - Google Search - Brave"
DEBUG: Starting monitor
DEBUG: Executing: copyq --clipboard-access monitorClipboard
DEBUG: Clipboard monitoring enabled.
DEBUG: Raising window "copyq not saving clipboard - Google Search - Brave"
DEBUG: Focus window is "copyq not saving clipboard - Google Search - Brave"
DEBUG: Executing: Command "Show/hide main window"
DEBUG: Focus window is " _hh_ "
DEBUG: Raising window "CopyQ"
DEBUG: Geometry: Window "MainWindow": Restore geometry "Options/MainWindow_geometry": 645x421,673,343 -> 645x421,267,301
DEBUG: Saving items on application state change (2)
DEBUG: Focus window is " _hh_ "
DEBUG: Focus window is "New Issue · hluk/CopyQ - Brave"
DEBUG: Raising window "New Issue · hluk/CopyQ - Brave"

`

**Comment by hluk** [comment]
Looks like duplicate of #1505. Was it closed by a mistake?

**Comment by arbach** [comment]
No, I closed that and opened clean version here since we know what is the issue. so we don't have people going through unnecessary details.

**Comment by arbach** [comment]
Bump.

**Comment by garytechlead** [comment]
I am having the same issue

**Comment by arbach** [comment]
This issue still exist with v 5.0, clipboard is not registered in copyq.

**Comment by arbach** [comment]
@hluk   any clues on how to debug this further ? 

copyq spontaneously stop storing data copied to clipboard, but then start working again. maybe between OS suspends.

**Comment by arbach** [comment]
Screen shoot from 
`watch -n0.1 tail copyq.log`
![error](http://screenshot.studio/Oct02Zjk3Mz.png)

With this nothing is stored in copyq

**Comment by hluk** [comment]
What's your desktop environment and window manager? Are you using Wayland instead of X11?

Is this perhaps happening only when copying from a specific app?

**Comment by arbach** [comment]
Ubuntu 20.04 LTS with x11. 

cat /etc/X11/default-display-manager
/usr/sbin/gdm3

loginctl show-session 1 -p Type
Type=x11

copyq either works everywhere, or doesn't work at all anywhere. I am suspecting this happen between suspends. Randomly when I bring laptop out of suspends, copyq works or doesn't work. but mostly doesn't work.

It always works after restart.

**Comment by arbach** [comment]
Something is not starting properly after suspension sometimes.  But even if I kill and start again it doesn't work. The only sure way to make work is to restart the machine. or keep trying suspending multiple times until it works.

Is there another way to restart ? @hluk 

This is what I tried and did not work

`copyq exit`
`export COPYQ_LOG_LEVEL='DEBUG'`
`export COPYQ_LOG_FILE="/tmp/copyq.log"`
`copyq`


or with killall 

if it's not working there is no way to make it work unless I restart the machine.

**Comment by hluk** [comment]
I'm guessing this is a bug in Qt. There is a similar issues already filed: https://bugreports.qt.io/browse/QTBUG-71961

**Comment by arbach** [comment]
what is an effective way to restart as work around ? restarting copyq did not resolve the issue.

**Comment by hluk** [comment]
> restarting copyq did not resolve the issue.

Can you install most recent version (5.0.0)? It can be installed from PPA: https://copyq.readthedocs.io/en/latest/installation.html

Also, please check if the app can still access current clipboard content - you can display the clipboard with Ctrl+Shift+C from the main application window.

**Comment by arbach** [comment]
I am already on 5.0.0

Yes, app can access the content, but it's not stored. 

![test content](http://screenshot.studio/Oct18MDNmZD.png)

**Comment by arbach** [comment]
Is there anything I can do to make it work other than restarting the whole machine ?

**Comment by Hrishio** [comment]
Does this issue has some solid solution ?

**Comment by Chealer** [comment]
> I am having the same issue

@garytechlead: Whenever you indicate you are affected by a behavioral issue, please provide a description of your environment, minimally indicating your version.
Do you get the same message as @arbach?

**Comment by Chealer** [comment]
Thank you for reporting @arbach 
Can you indicate a few applications where copy fails? For example, does it affect Mozilla Firefox?
Please replace the fourth step of your list of reproduction steps with an “Actual behavior” section which clearly indicates what occurs; is the effect only on CopyQ, or does this generally break X11 clipboards?

I am a little confused by your report; you mention the message “Clipboard unchanged, selection unchanged. Test again in 0ms.”, but it does not show in your CopyQ logs?🤨

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>
